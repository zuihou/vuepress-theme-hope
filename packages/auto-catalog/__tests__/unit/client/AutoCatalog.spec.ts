/* eslint-disable @typescript-eslint/naming-convention */
import { mount } from "@vue/test-utils";
import { describe, expect, it, vi } from "vitest";
import { type MockGlobalOption, resetSetup, setup } from "vuepress-test-utils";

import { locales } from "../../../src/node/locales.js";

const mockRoutes = [
  {
    path: "/",
    meta: {
      title: "Home",
    },
  },
  ...["a", "b", "c"]
    .map((item) => ({
      path: `/${item}.html`,
      meta: {
        title: `Title ${item.toUpperCase()}`,
      },
    }))
    .flat(1),
  ...["a", "b", "c"]
    .map((item1) =>
      ["a", "b", "c"].map((item2) => ({
        path: `/${item1}/${item2}.html`,
        meta: {
          title: `Title ${item1.toUpperCase()}-${item2.toUpperCase()}`,
        },
      }))
    )
    .flat(2),
  ...["a", "b", "c"]
    .map((item1) =>
      ["a", "b", "c"].map((item2) =>
        ["a", "b", "c"].map((item3) => ({
          path: `/${item1}/${item2}/${item3}.html`,
          meta: {
            title: `Title ${item1.toUpperCase()}-${item2.toUpperCase()}-${item3.toUpperCase()}`,
          },
        }))
      )
    )
    .flat(3),
];

const globals: MockGlobalOption[] = [
  ["AUTO_CATALOG_LOCALES", { "/": locales["/en/"] }],
  ["AUTO_CATALOG_INDEX_META_KEY", "index"],
  ["AUTO_CATALOG_TITLE_META_KEY", "title"],
  ["AUTO_CATALOG_ICON_META_KEY", "icon"],
  ["AUTO_CATALOG_ORDER_META_KEY", "order"],
];

describe("AutoCatalog", () => {
  it("should render empty hint", async () => {
    const mocks = setup(vi, {
      globals,
    });

    Object.entries(mocks).forEach(([module, factory]) => {
      vi.doMock(module, factory);
    });

    const AutoCatalog = (
      await import("../../../src/client/components/AutoCatalog.js")
    ).default;

    const wrapper = mount(AutoCatalog);

    expect(wrapper.html()).toMatchSnapshot();

    resetSetup(vi);
  });

  it("should render catalog with level2", async () => {
    const mocks = setup(vi, {
      routes: mockRoutes,
      globals,
    });

    Object.entries(mocks).forEach(([module, factory]) => {
      vi.doMock(module, factory);
    });

    const AutoCatalog = (
      await import("../../../src/client/components/AutoCatalog.js")
    ).default;

    const wrapper = mount(AutoCatalog);

    expect(wrapper.html()).toMatchSnapshot();

    resetSetup(vi);
  });
});
