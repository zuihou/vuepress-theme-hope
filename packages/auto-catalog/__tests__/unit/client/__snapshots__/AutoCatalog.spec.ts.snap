// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AutoCatalog > should render catalog with level2 1`] = `
"<div class=\\"auto-catalog-wrapper\\">
  <h2 class=\\"main-title\\">Catalog</h2>
  <h3 id=\\"Title A\\" class=\\"child-title\\"><a href=\\"#Title A\\" class=\\"header-anchor\\">#</a>
    <router-link class=\\"catalog-title\\" to=\\"/a.html\\"></router-link>
  </h3>
  <!---->
  <h3 id=\\"Title B\\" class=\\"child-title\\"><a href=\\"#Title B\\" class=\\"header-anchor\\">#</a>
    <router-link class=\\"catalog-title\\" to=\\"/b.html\\"></router-link>
  </h3>
  <!---->
  <h3 id=\\"Title C\\" class=\\"child-title\\"><a href=\\"#Title C\\" class=\\"header-anchor\\">#</a>
    <router-link class=\\"catalog-title\\" to=\\"/c.html\\"></router-link>
  </h3>
  <!---->
</div>"
`;

exports[`AutoCatalog > should render empty hint 1`] = `
"<div class=\\"auto-catalog-wrapper\\">
  <h2 class=\\"main-title\\">Catalog</h2>
  <p class=\\"empty-catalog\\">No catalog</p>
</div>"
`;
